packages/shared/src/components/post/PostItemCard.tsx

```diff
@@ -39,43 +44,29 @@ export default function PostItemCard({
   onHide,
   className,
   onContextMenu,
+  showVoteActions = false,
+  analyticsOrigin = Origin.Feed,
 }: PostItemCardProps): ReactElement {
+  const { user } = useContext(AuthContext);
   const { timestampDb, post } = postItem;
-  const { isFeedbackEnabled } = usePostFeedback({ post });
   const onHideClick = (e: MouseEvent) => {
     e.stopPropagation();
     onHide({ postId: post.id, timestamp: timestampDb });
   };
   const article = post?.sharedPost ?? post;
-  const { updatePost } = useUpdatePost();
 
-  const onUpvotePostMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.upvote(post),
-  });
-  const onCancelPostUpvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.cancelUpvote(post),
-  });
-  const onDownvotePostMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.downvote(post),
-  });
-  const onCancelPostDownvoteMutate = updatePost({
-    id: post.id,
-    update: mutationHandlers.cancelDownvote(post),
-  });
+  const client = useQueryClient();
+  const readingHistory = client.getQueryData<ReadHistoryInfiniteData>(
+    generateQueryKey(RequestKey.ReadingHistory, user),
+  );
```

Comment:
Why do we interact with reading history on the postItemCard?

Reply:
The component is used for history list and we need the data to manipulate the reading history cache.

Now that you said it I can probably move this inside the `useReadHistoryVotePost` hook.

Reply:
Updated here aab68d6842f9846bb3e1f75330c19d742b143a30